client.go

```diff
@@ -185,34 +183,15 @@ func (c *Client) UpdateTrait(user User, toUpdate *Trait) (*Trait, error) {
 	return trait, err
 }
 
-func (c *Client) UpdateTraits(user User, object interface{}) ([]*Trait, error) {
-	value := reflect.ValueOf(object)
-	typ := reflect.TypeOf(object)
-	var bulk []Trait
-	for i := 0; i < value.NumField(); i++ {
-		field := value.Field(i)
-		toUpdate := Trait{}
-		switch field.Kind() {
-		case reflect.Bool:
-			toUpdate.Value = strconv.FormatBool(field.Bool())
-		case reflect.String:
-			toUpdate.Value = field.String()
-		case reflect.Int:
-			toUpdate.Value = strconv.FormatInt(field.Int(), 10)
-		case reflect.Float32, reflect.Float64:
-			toUpdate.Value = strconv.FormatFloat(field.Float(), 'f', -1, 64)
-		default:
-			return nil, fmt.Errorf("invalid type of field '%s' (bool, int and string are supported)", typ.Field(i).Name)
-		}
-		toUpdate.Identity = user
-		toUpdate.Key = typ.Field(i).Name
-		bulk = append(bulk, toUpdate)
+// UpdateTraits updates trait values in bulk. Any set identity is replaced with the provided user.
+func (c *Client) UpdateTraits(user User, traits []*Trait) ([]*Trait, error) {
```

Comment:
This change is not backward compatible. Version that operates on object was created on user request, so we can't remove this logic.

Instead of changing interface I would suggest adding "special case" for `[]*Trait` (maybe also `[]Trait`). Such change would be backward-compatible and convenient for developers.

Reply:
Thank you for the review.

I've added in a type switch in the function and restored the previous functionality and tests. Support for both `[]*Trait` and `[]Trait` is there.

Unfortunately I had to copy-paste a small chunk of code (handling is identical for `[]*Trait` and `[]Trait`).